proto/trip.proto

```diff
@@ -325,6 +325,7 @@ message TripLeg {
   optional BoundingBox bbox = 9;
   optional ShapeAttributes shape_attributes = 10;
   repeated Incident incidents = 11;
+  repeated string algorithms = 12;
```

Comment:
this is just a nice way to get insight into what algorithms were used to make a leg. we could live without it but its good book keeping and useful for integration style unitesting